File: week01-210619-210625/moregorenine/stages/s1_IO_and_arithmetic_operations/q11_a_multiply_b/main.js
const fs = require('fs')
const stdin = (process.platform === 'linux'
        ? fs.readFileSync('/dev/stdin').toString()
        : `472
385`
).split('\n')

const number1 = parseInt(stdin[0], 10)
const number2 = parseInt(stdin[1], 10)
const countTen = 1
const total = 0

printResult(number1, number2, countTen, total)

function printResult(number1, number2, countTen, total) {
    let digit = number2 % 10
    let rest = Math.floor(number2 / 10)
    if (rest >= 0) {
        let result = number1 * digit
        console.log(result)
        total += result * countTen
        countTen = countTen * 10
    }

    if (rest > 0) {
        printResult(number1, rest, countTen, total)
    } else {
        console.log(total)
    }
}

File: week01-210619-210625/moregorenine/stages/s1_IO_and_arithmetic_operations/q11_a_multiply_b/Main.java
package kr.boj.stages.s1_IO_and_arithmetic_operations.q11_a_multiply_b;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int number1 = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        int number2 = Integer.parseInt(st.nextToken());
        int countTen = 1;
        int total = 0;

        printResult(number1, number2, countTen, total);

    }

    private static void printResult(int number1, int number2, int countTen, int total) {
        int digit = number2 % 10;
        int rest = number2 / 10;
        if (rest >= 0) {
            int result = number1 * digit;
            System.out.println(result);
            total += result * countTen;
            countTen = countTen * 10;
        }

        if (rest > 0) {
            printResult(number1, rest, countTen, total);
        } else {
            System.out.println(total);
        }
    }

}
